### PR TITLE
Add mitmproxy certificate authority to Docker configuration

### PR DESCRIPTION
**Pull request by remi**
Like #5, let’s add another configuration for another tool 😅 ([successfully tested here](https://github.com/mirego/elixir-boilerplate/actions/runs/5404058499/jobs/9817731997#step:9:21)).

Docker supports custom certificates in `/etc/docker/certs.d/<host>:<port>` ([reference](https://docs.docker.com/engine/security/certificates/#understand-the-configuration)).

GitHub-provided macOS runners [do not have Docker running](https://github.com/search?q=repo%3Aactions%2Frunner-images+docker+path%3Amacos&type=code) so it’s only for Ubuntu runners.

### CODE REVIEW

File: monitor/setup.sh [hunk 1]
```diff
@@ -116,6 +116,10 @@ elif [ "$RUNNER_OS" = "Linux" ]; then
   echo "REQUESTS_CA_BUNDLE=/home/mitmproxyuser/.mitmproxy/mitmproxy-ca-cert.pem" >> $GITHUB_ENV
   # set environment variable for the Elixir Hex package manager to use the certificate
   echo "HEX_CACERTS_PATH=/home/mitmproxyuser/.mitmproxy/mitmproxy-ca-cert.pem" >> $GITHUB_ENV
+  # copy certificate to Docker directory
+  sudo mkdir -p /etc/docker/certs.d/registry-1.docker.io:443
+  sudo cp ~/mitmproxy-ca-cert.crt /etc/docker/certs.d/registry-1.docker.io:443
```
Comment (remi) [inline monitor/setup.sh:121]: Well I just realized that `registry-1.docker.io:443` might have to be provided as an input value, right? 😕

**Comment by JarLob** [comment]
Installing CA in the upper directory didn't work out? :(

**Comment by remi** [comment]
> Installing CA in the upper directory didn't work out? :(

Unfortunately no 😞 It looks like a lot of tools out there have their own way of handling CAs…

**Comment by JarLob** [comment]
I think your changes are good, but I tried to understand if docker doesn't work for everyone, so I have created a test workflow and docker pull and docker build do not fail even without the fix https://github.com/GitHubSecurityLab/actions-permissions/actions/runs/5411765564/jobs/9834933823

Could you help me to understand what is missing to reproduce?

**Pull request closed without merging**